libs/libcds/include/hpx/libcds/hpx_tls_manager.hpp

```diff
@@ -0,0 +1,250 @@
+//  Copyright (c) 2020 Weile Wei
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0.
+//  (See accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#ifndef LIBCDS_HPX_TLS_MANAGER
+#define LIBCDS_HPX_TLS_MANAGER
+
+#include <hpx/config.hpp>
+#include <hpx/errors/exception.hpp>
+#include <hpx/runtime_local/config_entry.hpp>
+//
+#include <cds/gc/details/hp_common.h>
+#include <cds/gc/hp.h>
+#include <cds/init.h>
+#include <cds/threading/details/cxx11_manager.h>
+#include <cds/urcu/general_buffered.h>
+
+#include <atomic>
+#include <cstddef>
+#include <iostream>
+#include <string>
+
+/// \cond NODETAIL
+namespace cds { namespace gc { namespace hp { namespace details {
+
+    class HPXDataHolder
+    {
+    public:
+        CDS_EXPORT_API static CDS_EXPORT_API thread_data* getTLS();
```

Comment:
Two questions:
1. Should this actually be using `HPX_EXPORT` instead?
2. Can the code be in `hpx::cds` instead of `cds::...`? This is in the end our code, not libcds code.

Reply:
For 1, in LibCDS code, it requires `CDS_EXPORT_API` so I was following their style.
For 2, @biddisco maybe have more ideas about it.

Reply:
Do we need the macro twice? Also, whether we should be using `HPX_EXPORT` or `CDS_EXPORT_API` depends on what shared library the implementation of this function ends up being linked to.